astra-sim/workload/Workload.cc

```diff
@@ -170,69 +175,97 @@ void Workload::issue_comp(shared_ptr<Chakra::ETFeederNode> node) {
         ChakraCollectiveCommType::ALL_REDUCE) {
       DataSet* fp = sys->generate_all_reduce(
-          node->getChakraNode()->comm_size(),
+          node->getChakraNodeFlattened()->comm_size(),
           involved_dim,
           comm_group,
-          node->getChakraNode()->comm_priority());
+          node->getChakraNodeFlattened()->comm_priority());
       collective_comm_node_id_map[fp->my_id] = node->getChakraNode()->id();
+      collective_comm_wrapper_map[fp->my_id] = fp;
       fp->set_notifier(this, EventType::CollectiveCommunicationFinished);
 
     } else if (
-        node->getChakraNode()->comm_type() ==
+        node->getChakraNodeFlattened()->comm_type() ==
         ChakraCollectiveCommType::ALL_TO_ALL) {
       DataSet* fp = sys->generate_all_to_all(
-          node->getChakraNode()->comm_size(),
+          node->getChakraNodeFlattened()->comm_size(),
           involved_dim,
           comm_group,
-          node->getChakraNode()->comm_priority());
+          node->getChakraNodeFlattened()->comm_priority());
       collective_comm_node_id_map[fp->my_id] = node->getChakraNode()->id();
+      collective_comm_wrapper_map[fp->my_id] = fp;
       fp->set_notifier(this, EventType::CollectiveCommunicationFinished);
 
     } else if (
-        node->getChakraNode()->comm_type() ==
+        node->getChakraNodeFlattened()->comm_type() ==
         ChakraCollectiveCommType::ALL_GATHER) {
       DataSet* fp = sys->generate_all_gather(
-          node->getChakraNode()->comm_size(),
+          node->getChakraNodeFlattened()->comm_size(),
           involved_dim,
           comm_group,
-          node->getChakraNode()->comm_priority());
+          node->getChakraNodeFlattened()->comm_priority());
       collective_comm_node_id_map[fp->my_id] = node->getChakraNode()->id();
+      collective_comm_wrapper_map[fp->my_id] = fp;
       fp->set_notifier(this, EventType::CollectiveCommunicationFinished);
 
     } else if (
-        node->getChakraNode()->comm_type() ==
+        node->getChakraNodeFlattened()->comm_type() ==
         ChakraCollectiveCommType::REDUCE_SCATTER) {
       DataSet* fp = sys->generate_reduce_scatter(
-          node->getChakraNode()->comm_size(),
+          node->getChakraNodeFlattened()->comm_size(),
           involved_dim,
           comm_group,
-          node->getChakraNode()->comm_priority());
+          node->getChakraNodeFlattened()->comm_priority());
       collective_comm_node_id_map[fp->my_id] = node->getChakraNode()->id();
+      collective_comm_wrapper_map[fp->my_id] = fp;
       fp->set_notifier(this, EventType::CollectiveCommunicationFinished);
-    }
-  } else if (
-      node->getChakraNode()->node_type() == ChakraNodeType::COMM_SEND_NODE) {
+    } else if (
+        node->getChakraNodeFlattened()->comm_type() ==
+        ChakraCollectiveCommType::BROADCAST) {
+      //broadcast colelctive has not been implemented in ASTRA-SIM yet. So, we just use its real system mesurements
```

Comment:
What makes the broadcast node difficult to implement?

Can we generate a warning here that is printed to the terminal?